src/v3/src/transformer/layout/consent/transformAdminConsent.ts

```diff
@@ -0,0 +1,145 @@
+      scopes,
+    } = {},
+  } = transaction;
+
+  uischema.elements = removeUIElementWithName('consent', uischema.elements);
+
+  const descriptionEle: DescriptionElement = {
+    type: 'Description',
+    contentType: 'subtitle',
+    options: {
+      content: loc('oie.consent.scopes.granular.description', 'login'),
+      dataSe: 'consent-description',
+    },
+  };
+  uischema.elements.push(descriptionEle);
+
+  if (Array.isArray(scopes)) {
+    const scopesByGroup = groupBy(scopes as ConsentScope[], (scope: ConsentScope) => {
+      const [,groupKey] = scope.name.split('.');
+      return SCOPE_GROUP_CONFIG[groupKey] ?? 'system';
+    });
+
+    Object.entries(scopesByGroup).forEach(([key, consentScopes]) => {
+      const heading: HeadingElement = {
+        type: 'Heading',
+        noMargin: true,
+        options: {
+          level: 5,
+          visualLevel: 3,
+          content: loc(GROUP_KEY_TO_I18N_KEY[key as ConsentScopeGroup], 'login'),
+          dataSe: 'scope-group--header',
+        },
+      };
+      uischema.elements.push(heading);
+      uischema.elements.push(...consentScopes.map((scope) => {
+        const labelI18nKey = `consent.scopes.${scope.name}.label`;
+        const descrI18nKey = `consent.scopes.${scope.name}.desc`;
+        const isPredefinedLabel = doesI18NKeyExist(labelI18nKey);
+        const isPredefinedDesc = doesI18NKeyExist(descrI18nKey);
+        const label = isPredefinedLabel ? loc(labelI18nKey, 'login') : scope.label;
+        const description = isPredefinedDesc
+          ? loc(descrI18nKey, 'login')
+          : scope.desc;
+        const fieldName = `consent.scopes.${scope.name}`;
+        data[fieldName] = true;
+        return {
+          type: 'Field',
+          translations: [
+            {
+              name: 'label',
+              i18nKey: labelI18nKey,
+              value: label,
+              noTranslate: !isPredefinedLabel,
+            },
+            {
+              name: 'description',
+              i18nKey: descrI18nKey,
+              value: description,
+              noTranslate: !isPredefinedDesc,
+            },
+          ],
+          options: {
+            inputMeta: {
+              name: fieldName,
+              type: 'boolean',
+              mutable: false,
+            },
+          },
+        } as FieldElement;
+      }));
+    });
+  }
+
+  const submitButton: ButtonElement = {
+    type: 'Button',
+    label: loc('consent.required.consentButton', 'login'),
+    options: {
+      type: ButtonType.BUTTON,
+      includeData: false,
+      actionParams: { consent: true },
+      step: stepName!,
+      dataType: 'save',
+      variant: 'secondary',
+    },
+  };
+
+  const cancelButton: ButtonElement = {
+    type: 'Button',
+    label: loc('oform.cancel', 'login'),
+    options: {
+      type: ButtonType.BUTTON,
+      includeData: false,
+      actionParams: { consent: false },
+      step: stepName!,
```

Comment:
Could we avoid the non-null assertion here by doing a typecheck with early return at the top of the transform?

Something like
```ts
if (!stepName) {
  return formBag;
}
```